server/graphql/common/expenses.ts

```diff
@@ -531,6 +531,13 @@ export async function createExpense(
     throw new Unauthorized('Missing expense.collective.id');
   }
 
+  if (remoteUser.rolesByCollectiveId) {
```

Comment:
Why wouldn't that be set? Why skipping the check if it's not?

Reply:
@Betree : I just removed this condition. 😄 

I just added this check for good measure, in case if it isn't set we skip gracefully without throwing a null pointer in the next line. But even for a user who doesn't have any roles in any collective it seems to me that `rolesByCollectiveId` returns an empty object. So I think it's okay to remove it. 👍🏽

Reply:
@Betree : Actually I spoke too soon, it seems that when we remove it some tests are failing; https://github.com/opencollective/opencollective-api/runs/2401851404?check_suite_focus=true. 🤔

Reply:
`rolesByCollectiveId` is not supposed to be `null` here, as it's normally populated before entering any query or mutation. We can eventually add an `await remoteUser?.populateRoles()` in `makeRequest` if some tests are inconsistent on this regard: https://github.com/opencollective/opencollective-api/blob/2d46ef85b6894c690704fcc7b32e5527f34253f3/test/utils.js#L62

Reply:
@Betree : Done. It was the graphql tests that were failing and I had to add the `populateRoles()` in two places to make it work. 😄